06_puppy/undrewb/store_test.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type storerSuite struct {
+	suite.Suite
+	store      Storer
+	storerType func() Storer
+}
+
+func TestStorers(t *testing.T) {
+	suite.Run(t, &storerSuite{
+		storerType: func() Storer { return &MapStore{} },
+	})
+	suite.Run(t, &storerSuite{
+		storerType: func() Storer { return &SyncStore{} },
+	})
+}
+
+func (s *storerSuite) SetupTest() {
+	s.store = s.storerType()
+
+	corgi := &Puppy{1000, "corgi", "orange and white", "$$$$$$$"}
+	_ = s.store.CreatePuppy(corgi)
```

Comment:
I don't believe you should have `createPuppy` method inside `SetUpTest` func, as if your `createPuppy` method is wrong, then it would have failed here. Consider just have raw data in here.

Reply:
If I tie the insert to implementation I cant use the same code for testing different implementations of the same interface. If the test fails because the implementation is wrong thats a good thing :)